docs/_releases/v2.0.0-pre.4/matchers.md

```diff
@@ -112,6 +112,26 @@ Requires the value to be a `Function`.
 Requires the value to be an `Array`.
 
 
+#### `sinon.match.array.deepEquals(iterable)`
+
+Requires an iterable value to be deep equal another one.
```

Comment:
> expanding the namespace through match.map.*, match.set.*, etc

This is going to be a lot of duplicate or (hopefully) resused code, but I think that is the most explicit option.

> Agree/disagree, @fearphage?

So the first option being to make this only work for arrays since it's in the `array` namespace? If that is the case, I would agree. I think we should confirm that though. 

For instance, `matcher.array.startsWith(...)` should throw when you don't pass an array to it. It seems like you're doing 2 things at once here. `matcher.array()` && `array.startsWith()`. So if I pass `{length: 1, 0: 'foo'}`, it would fail saying the test expected an array. Are we in agreement or did I miss the mark?